### ISSUE TITLE
`ShellExecute{A,W}` and `SHELLEXECUTEINFO{A,W}` should use a unique type instead of `HINSTANCE`

### ISSUE THREAD

**Issue by ChrisDenton** [microsoft/win32metadata#1456]
E.g. see the documentation of the return value for [`ShellExecute`](https://learn.microsoft.com/en-us/windows/win32/api/shellapi/nf-shellapi-shellexecutew#return-value):

> The return value is cast as an HINSTANCE for backward compatibility with 16-bit Windows applications. It is not a true HINSTANCE, however. It can be cast only to an INT_PTR and compared to either 32 or the following error codes below.

The [`SHELLEXECUTEINFO`](https://learn.microsoft.com/en-us/windows/win32/api/shellapi/ns-shellapi-shellexecuteinfoa#members) case is a bit more complicated than that. The `hInstApp` member has a similar remark as above but also  `fMask` can set `SEE_MASK_FLAG_HINST_IS_SITE` which changes the meaning:

> SEE_MASK_FLAG_HINST_IS_SITE (0x08000000)
> The hInstApp member is used to specify the [IUnknown](https://learn.microsoft.com/en-us/windows/desktop/api/unknwn/nn-unknwn-iunknown) of an object that implements [IServiceProvider](https://learn.microsoft.com/en-us/previous-versions/windows/internet-explorer/ie-developer/platform-apis/cc678965(v=vs.85)). This object will be used as a site pointer. The site pointer is used to provide services to the [ShellExecute](https://learn.microsoft.com/en-us/windows/desktop/api/shellapi/nf-shellapi-shellexecutea) function, the handler binding process, and invoked verb handlers.

In short, `hInstApp` can be one of the `SE_ERR_` values, a value greater than 32 or a site pointer.

**Comment by ChrisDenton** [comment]
I think the `HINSTANCE` in these cases should be unique. So maybe this can be done through namespacing?

**Comment by kennykerr** [comment]
I'd like to avoid the same type appearing in different namespaces as that prevents any future roundtripping in C/C++. Given the `ShellExecute` comment, it sounds like it should just be `IntPtr` and the caller is responsible for casting to something more specific as needed. But `HINSTANCE`'s underying type is already `IntPtr`. 

@ChrisDenton what's the downside with just leaving it as is? Either way you need to read the docs and handle it carefully.

**Comment by ChrisDenton** [comment]
I guess leaving it as it is is ok. I've just been looking into automating some stronger typing. But if need be I can always add another layer of metadata for my needs.

**Comment by mikebattista** [comment]
We could introduce new special types for these APIs. Do you just want 2 new special types, one for each API here? The definitions would otherwise be identical to HINSTANCE.

**Comment by mikebattista** [comment]
Ok. That does seem to complicate things a bit if you want a dynamic definition like that. I'd prefer to keep as is.
